src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -43,6 +59,55 @@ function recursivelyAppendChildren(parent : Element, child : HostChildren<Instan
   }
 }
 
+function removeMissingPrevDOMProperties(
+  domElement: Element,
+  oldProps: Props,
+  newProps: Props,
+) {
+  for (const propKey in oldProps) {
+    if (newProps.hasOwnProperty(propKey) ||
+        !oldProps.hasOwnProperty(propKey) ||
+        oldProps[propKey] == null) {
+      continue;
+    }
+    // TODO: unregister events, remove styles, handle custom components
+    if (DOMProperties[propKey] || isCustomAttribute(propKey)) {
+      deleteValueForProperty(domElement, propKey, 0);
+    }
+  }
+}
+
+function applyNextDOMProperties(
+  domElement: Element,
+  oldProps: Props | null,
+  newProps: Props,
+) {
+  // TODO: associate DOM nodes with components for ReactPerf
```

Comment:
Trying to understand how some things work, it looks like the `ReactDOMComponentTree` logic was hoisted out of `ReactDOMProperties` as a stepping stone to *this* TODO.

If I understand correctly: had this change not hoisted `ReactDOMComponentTree` out of the `(set|delete)ValueForProperty` calls the tests would fail. With this change a subset of the tests and functionality pass, though the `ReactPerf` and `ReactDevtools` would still not be able to take advantage of this.

Is this an okay-ish understanding or am I off?